rfc2544/README.md

```diff
@@ -15,26 +15,53 @@ The container image build from this directory can be used for either automation
 
 To achieve higher traffic rate, the two trafficgen ports should come from different NICs.
 
-In general using pysical function (PF) as the trafficgen ports can have better throughput performance than using the virtual function (VF). Building the container image based on TRex version 2.88 is recommended; the only exception is to use VF on Intel E810 nic as the trafficgen ports - in this case, building the container image based on TRex version 3.02 is recommended.   
+In general using pysical function (PF) as the trafficgen ports can have better throughput performance than using the virtual function (VF). Building the container image based on TRex version 2.88 is recommended; the only exception is to use VF on Intel E810 nic as the trafficgen ports - in this case, building the container image based on TRex version 3.02 is recommended.
 
 ## Openshift integration demo
 
 [![Watch the video](https://img.youtube.com/vi/C5s9DZC3D6c/maxresdefault.jpg)](https://youtu.be/C5s9DZC3D6c)
 
+## Build the container image
+
+To build with the default TREX_VERSION,
+```
+podman build -t localhost/trafficgen .
+```
+
+To build with a different TREX_VERSION, say for the purpose of running it on the virtual function on E810,
+```
+podman build -t localhost/trafficgen --build-arg TREX_VERSION=3.02 .
+```
+
+## Bind trafficgen port to vfio-pci
+
+Before running the trafficgen container, the trafficgen ports need to be bound to vfio-pci first. Assume the RPM package `dpdk-tools` is pre-installed on the system, and `dpdk-devbind.py` is available with this RPM install. Here is the steps to bind the ports to the vfio-pci,
+```
+modprobe vfio-pci
+dpdk-devbind.py -u <port1_pci> <port2_pci>
+dpdk-devbind.py -b vfio-pci <port1_pci> <port2_pci>
+```
+
 ## Podman run example for manual test
 
-Ensure that for a background run the `CMD` field of the Dockerfile is properly set to `start`:
+First, identify the numa node associated with the NIC pic slot,
 ```
-CMD ["/root/trafficgen_entry.sh", "start"]
+cat /sys/bus/pci/devices/<pci_address>/numa_node
 ```
 
-Run the trafficgen
-`podman run -it --rm --privileged -v /dev:/dev -v /sys:/sys -v /lib/modules:/lib/modules --cpuset-cpus 4-11 -e pci_list=0000:03:00.0,0000:03:00.1 docker.io/cscojianzhan/trafficgen`
+Select a cpuset on this numa node.
+
+For manual test, users normally do not use pod, but run the trafficgen container directly,
+```
+podman run -it --rm --privileged -v /dev/hugepages:/dev/hugepages -v /sys/bus/pci/devices:/sys/bus/pci/devices -v /lib/modules:/lib/modules --cpuset-cpus 4-11 -e pci_list=0000:03:00.0,0000:03:00.1 localhost/trafficgen start
```

Comment:
We should probably use` --cpuset-cpus 5,7,9,11,13,15` in this command to illustrate our selection.